### PR TITLE
Backfill play tagline for NFT Catalog

### PR DESCRIPTION
**Pull request by zihehuang**
This PR Addresses https://github.com/dapperlabs/nba-smart-contracts/issues/167

# What I changed
- Added a tagline setter function to Play struct
- Added an access function for Admin resource to backfill the play struct with taglines
- Updated `description` to use the Play tagline if tagline exists

# How is this tested
- Added a test that asserts tagline being added to Play metadata

### CODE REVIEW

File: contracts/TopShot.cdc [hunk 1]
```diff
@@ -72,6 +72,8 @@ pub contract TopShot: NonFungibleToken {
 
     // Emitted when a new Play struct is created
     pub event PlayCreated(id: UInt32, metadata: {String:String})
+    // Emitted when a new Play's metadata is updated
+    pub event PlayUpdated(id: UInt32, metadata: {String:String})
```
Comment (jrkhan) [inline contracts/TopShot.cdc:76]: Don't think we need an event for this at present

File: contracts/TopShot.cdc [hunk 2]
```diff
@@ -696,6 +706,17 @@ pub contract TopShot: NonFungibleToken {
                 .concat(serialNumber)
         }
 
+        pub fun description(): String {
```
Comment (jrkhan) [inline contracts/TopShot.cdc:709]: need a descriptive comment for the method

File: contracts/TopShot.cdc [hunk 3]
```diff
@@ -174,6 +176,14 @@ pub contract TopShot: NonFungibleToken {
             self.playID = TopShot.nextPlayID
             self.metadata = metadata
         }
+
+        access(contract) fun updateTagline(tagline: String): UInt32 {
```
Comment (jrkhan) [inline contracts/TopShot.cdc:180]: we could use a descriptive comment for the method

**Comment by jrkhan** [comment]
I believe the intent will be to _remove_ the method that we use to backfill the tagline once it has been completed - it is being added temporarily.

File: contracts/TopShot.cdc [hunk 4]
```diff
@@ -696,6 +706,19 @@ pub contract TopShot: NonFungibleToken {
                 .concat(serialNumber)
         }
 
+        // The description of the Moment. If Tagline property of the play is empty, compose it using the buildDescString function
+        // If the Tagline property is not empty, use that as the description
```
Comment (satyamakgec) [inline contracts/TopShot.cdc:710]: To support code comments-based documentation, please use `///` instead of `//`.

File: contracts/TopShot.cdc [hunk 5]
```diff
@@ -943,6 +966,16 @@ pub contract TopShot: NonFungibleToken {
             return newID
         }
 
+        // Temporarily enabled so the description of the play can be backfilled
+        // Parameters: playID: The ID of the play to update
+        //             tagline: A string to be used as the tagline for the play
+        // Returns: The ID of the play
```
Comment (satyamakgec) [inline contracts/TopShot.cdc:972]: use `///` instead of `//`

File: transactions/admin/update_tagline.cdc [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        for key in plays.keys {
+            self.adminRef.updatePlayTagline(playID: key, tagline: plays[key] ?? panic("No tagline for play"))
+        }
+    }
+
+    post {
+        TopShot.getPlayMetaDataByField(playID: self.firstKey, field: "Tagline") != nil:
+            "first play's tagline doesnt exist"
+        TopShot.getPlayMetaDataByField(playID: self.lastKey, field: "Tagline") != nil:
+            "last play's tagline doesnt exist"
```
Comment (satyamakgec) [inline transactions/admin/update_tagline.cdc:37]:
```suggestion
            "Last play's tagline does not exist"
```

File: transactions/admin/update_tagline.cdc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    execute {
+        // update multiple plays with the specified metadata
+        for key in plays.keys {
+            self.adminRef.updatePlayTagline(playID: key, tagline: plays[key] ?? panic("No tagline for play"))
+        }
+    }
+
+    post {
+        TopShot.getPlayMetaDataByField(playID: self.firstKey, field: "Tagline") != nil:
+            "first play's tagline doesnt exist"
```
Comment (satyamakgec) [inline transactions/admin/update_tagline.cdc:35]:
```suggestion
            "First play's tagline does not exist"
```

**Review by satyamakgec — COMMENTED** [review]
LGTM, some minor comments

File: contracts/TopShot.cdc [hunk 8]
```diff
@@ -696,6 +706,19 @@ pub contract TopShot: NonFungibleToken {
                 .concat(serialNumber)
         }
 
+        /// The description of the Moment. If Tagline property of the play is empty, compose it using the buildDescString function
+        /// If the Tagline property is not empty, use that as the description
+        pub fun description(): String {
+            let playDesc: String? = TopShot.getPlayMetaDataByField(playID: self.data.playID, field: "Tagline")
```
Comment (jrkhan) [inline contracts/TopShot.cdc:712]: let playDesc: String = TopShot.getPlayMetaDataByField(playID: self.data.playID, field: "Tagline") ?? ""
if playDesc.length > 0 {
   return playDesc
}
return self.buildDescString()

**Review by jrkhan — APPROVED** [review]

**Pull request merged**